pkg/controller/pxc/controller_test.go

```diff
@@ -57,3 +69,450 @@ var _ = Describe("PerconaXtraDB Cluster", Ordered, func() {
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("controller should remove ssl-secrets", func() {
+				secret := &corev1.Secret{}
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, types.NamespacedName{
+						Namespace: cr.Namespace,
+						Name:      cr.Spec.SSLSecretName,
+					}, secret)
+
+					return k8serrors.IsNotFound(err)
+				}, time.Second*15, time.Millisecond*250).Should(BeTrue())
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, types.NamespacedName{
+						Namespace: cr.Namespace,
+						Name:      cr.Spec.SSLInternalSecretName,
+					}, secret)
+
+					return k8serrors.IsNotFound(err)
+				}, time.Second*15, time.Millisecond*250).Should(BeTrue())
+
+				Eventually(func() bool {
+					err := k8sClient.Get(ctx, types.NamespacedName{
+						Namespace: cr.Namespace,
+						Name:      cr.Name + "-ca-cert",
+					}, secret)
+
+					return k8serrors.IsNotFound(err)
+				}, time.Second*15, time.Millisecond*250).Should(BeTrue())
+			})
+
+			It("controller should delete issuers and certificates", func() {
+				issuers := &cm.IssuerList{}
+				Eventually(func() bool {
+
+					opts := &client.ListOptions{Namespace: cr.Namespace}
+					err := k8sClient.List(ctx, issuers, opts)
+
+					return err == nil
+				}, time.Second*30, time.Millisecond*250).Should(BeTrue())
+
+				Expect(issuers.Items).Should(BeEmpty())
+
+				certs := &cm.CertificateList{}
+				Eventually(func() bool {
+
+					opts := &client.ListOptions{Namespace: cr.Namespace}
+					err := k8sClient.List(ctx, certs, opts)
+
+					return err == nil
+				}, time.Second*30, time.Millisecond*250).Should(BeTrue())
+
+				Expect(certs.Items).Should(BeEmpty())
+			})
+		})
+	})
+})
+
+var _ = Describe("Finalizer delete-proxysql-pvc", Ordered, func() {
+	ctx := context.Background()
+
+	const crName = "del-proxysql-pvc-fnlz"
+	const ns = "del-proxysql-pvc-fnlz"
+	crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+
+		_, err = envtest.InstallCRDs(cfg, envtest.CRDInstallOptions{
+			Paths: []string{filepath.Join("testdata", "cert-manager.yaml")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context("delete-proxysql-pvc finalizer specified", Ordered, func() {
+
+		cr, err := readDefaultCR(crName, ns)
+		cr.Finalizers = append(cr.Finalizers, "delete-proxysql-pvc")
+		cr.Spec.SecretsName = "cluster1-secrets"
+		cr.Spec.HAProxy.Enabled = false
+		cr.Spec.ProxySQL.Enabled = true
+		os.Setenv("DISABLE_TELEMETRY", "true")
+
+		sfsWithOwner := appsv1.StatefulSet{}
+		sfs := statefulset.NewProxy(cr)
```

Comment:
Do you need to call this by hand, will the controller call it?

Reply:
I see that we should create it, but I didn't catch how can I access it.